### ISSUE TITLE
Add libvirt to run virtual machines

### ISSUE THREAD

**Issue by joshiste** [actions/runner-images#7670]
### Tool name

libvirt

### Tool license

LGPL

### Add or update?

- [X] Add
- [ ] Update

### Desired version

latest stable

### Brief description of tool

libvirt is an open-source API, daemon and management tool for managing platform virtualization. It can be used to manage KVM, Xen, VMware ESXi, QEMU and other virtualization technologies

### URL for tool's homepage

https://libvirt.org

### Platforms where you need the tool

- [ ] Azure DevOps
- [X] GitHub Actions

### Runner images where you need the tool

- [X] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Can this tool be installed during the build?

```bash
- name: Enable KVM group perms
        run: |
          echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
          sudo udevadm control --reload-rules
          sudo udevadm trigger --name-match=kvm
          sudo apt-get update
          sudo apt-get install -y libvirt-clients libvirt-daemon-system libvirt-daemon virtinst bridge-utils qemu qemu-system-x86
          sudo usermod -a -G kvm,libvirt $USER
```

**Comment by joshiste** [comment]
cc @nebuk89

**Comment by shamil-mubarakshin** [comment]
Hi @joshiste, we will consider this tool.

**Comment by DaanDeMeyer** [comment]
Can we split this into two? Adding the runner user to the kvm group is something that can be done ahead of time already as it unblocks existing use cases using qemu. It'd be great if users would not have to wait until libvirt is ready to be bundled just to get the kvm user added to the runner user's default groups.

**Comment by DaanDeMeyer** [comment]
Specifically, `usermod -a -G kvm runner` does not work because you have to start a new session to reflect the new group which doesn't seem to happen inbetween steps in Github Actions.

**Comment by mikhailkoliada** [comment]
Hey! As current azure linux sku we are using do not support nested virtualisation we can not pre-install it to the images as we can not guarantee that the tool will properly function at all, but it clearly can / should be used on [Larger Runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners) as it fully supports nested. As soon as nested is available on the standard runners the ll the qemu/kvm / libvirt stuff will be available as a part of the image, but ETA is unclear (my initial though was it would be available a little bit earlier). Closing as stale for now as we can not satisfy the request at least now.
@nebuk89 if you have more questions / ideas feel free to reach out to me directly.

**Comment by DaanDeMeyer** [comment]
@mikhailkoliada Can we please add the runner user to the `kvm` group independent of actually enabling libvirt and `kvm`? This is independent of `kvm` and also required for using /dev/vhost-vsock which is completely independent of KVM and already works on standard runners today.

**Comment by sebrandon1** [comment]
Are there any workarounds to having to reload your terminal after running `usermod -a -G kvm runner`?  I think I've tried everything but my user does not get added to the `libvirt` group.

**Comment by JojOatXGME** [comment]
@mikhailkoliada Would you re-open the ticket as nested virtualization now seems to be available for all standard runners?

Btw, I also want to highlight that making the user part of the `kvm` group seems much more important to me, than installing libvirt by default. This is why I find it strange that #8542 was closed as a duplicate. There are a lot more use-cases for nested virtualization, than libvirt. I think libvirt may even be a rather rare use-case. It seems quite natural that people who want to use libvirt, might need to install it. However, the workaround for not being part of the `kvm` group is quite hacky and people have already implemented it incorrectly. See #8670 as an example. If I remember correctly, even [the announcement for nested virtualization in GitHub's own blog](https://github.blog/changelog/2023-02-23-hardware-accelerated-android-virtualization-on-actions-windows-and-linux-larger-hosted-runners/) made the exact same error at the beginning, until it was pointed out by the community and fixed in the blog post. I also don't see any reason not to add the user to the `kvm` group, even if the runner does not support nested virtualization.
